src/Traits/ReflectionFunctionLikeTrait.php

```diff
@@ -181,18 +230,32 @@ public function getParameters()
     public function getReturnType()
     {
         $isBuiltin  = false;
-        $returnType = $this->functionLikeNode->getReturnType();
-        $isNullable = $returnType instanceof NullableType;
+        if ($this->functionLikeNode) {
+            $returnType = $this->functionLikeNode->getReturnType();
+            $isNullable = $returnType instanceof NullableType;
 
-        if ($isNullable) {
-            $returnType = $returnType->type;
-        }
-        if (is_object($returnType)) {
-            $returnType = $returnType->toString();
-        } elseif (is_string($returnType)) {
-            $isBuiltin = true;
+            if ($isNullable) {
+                $returnType = $returnType->type;
+            }
+            if (is_object($returnType)) {
+                $returnType = $returnType->toString();
+            } elseif (is_string($returnType)) {
+                $isBuiltin = true;
+            } else {
+                return null;
+            }
         } else {
-            return null;
+            if (PHP_VERSION_ID < 70000) {
```

Comment:
This check won't help you, because we can use PHP5 to parse PHP7 files and vice-versa.

Reply:
If we're in this else block there is no PHP source code to analyze, and we're only querying the PHP binary and any loaded extensions.